src/crawlee/base_storage_client/_base_key_value_store_client.py

```diff
@@ -114,3 +114,14 @@ async def delete_record(self, key: str) -> None:
         Args:
             key: The key of the record which to delete
         """
+
+    @abstractmethod
+    def get_public_url(self, key: str) -> str:
```

Comment:
Minor thing, but could you make this an `async` method? It's for forward compatibility in case some resource client needs to e.g., make a pre-signed S3 URL.

Reply:
Updated the method to be async